x86: library-check: 'error: SSE vector return without SSE enabled changes the ABI'

When building CBMC version 5.95.1 on 32-bit x86 with `-msse2 -mfpmath=sse` library-check fails with
```
[00:01:02] [ 46%] Generating library-check.stamp
...
[00:01:02] __libcheck.c: In function 's__builtin_ia32_psubsw128':
[00:01:02] __libcheck.c:219:1: error: SSE vector return without SSE enabled changes the ABI [-Werror=psabi]
[00:01:02]   219 | {
[00:01:02]       | ^
[00:01:02] __libcheck.c: In function 's__builtin_ia32_paddsw':
[00:01:02] __libcheck.c:248:1: error: MMX vector return without MMX enabled changes the ABI [-Werror=psabi]
[00:01:02]   248 | {
[00:01:02]       | ^
[00:01:02] __libcheck.c: In function 's__builtin_ia32_vec_ext_v4hi':
[00:01:02] __libcheck.c:160:7: error: MMX vector argument without MMX enabled changes the ABI [-Werror=psabi]
[00:01:02]   160 | short s__builtin_ia32_vec_ext_v4hi(__gcc_v4hi vec, int offset)
[00:01:02]       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
[00:01:02] __libcheck.c: In function 's__builtin_ia32_vec_ext_v8hi':
[00:01:02] __libcheck.c:169:7: note: the ABI for passing parameters with 16-byte alignment has changed in GCC 4.6
[00:01:02]   169 | short s__builtin_ia32_vec_ext_v8hi(__gcc_v8hi vec, int offset)
[00:01:02]       |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
[00:01:02] __libcheck.c:169:7: error: SSE vector argument without SSE enabled changes the ABI [-Werror=psabi]
```
This seems because while everything is compiled with `-msse2 -mfpmath=sse`  these flags are lost in `library_check.sh`.

CBMC version: 5.95.1
Operating system: ALT Linux
Exact command line resulting in the issue:
```
+ cmake -DCMAKE_SKIP_INSTALL_RPATH:BOOL=yes '-DCMAKE_C_FLAGS:STRING=-pipe -frecord-gcc-switches -Wall -fdiagnostics-color=always -fdiagnostics-color=always -fdiagnostics-color=always -g -O2  -march=i586 -mtune=generic -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DMINISAT_CONSTANTS_AS_MACROS -Wno-error=odr -msse2 -mfpmath=sse' '-DCMAKE_CXX_FLAGS:STRING=-pipe -frecord-gcc-switches -Wall -fdiagnostics-color=always -fdiagnostics-color=always -fdiagnostics-color=always -g -O2  -march=i586 -mtune=generic -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DMINISAT_CONSTANTS_AS_MACROS -Wno-error=odr -msse2 -mfpmath=sse' '-DCMAKE_Fortran_FLAGS:STRING=-pipe -frecord-gcc-switches -Wall -fdiagnostics-color=always -fdiagnostics-color=always -fdiagnostics-color=always -g -O2  -march=i586 -mtune=generic -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DMINISAT_CONSTANTS_AS_MACROS -Wno-error=odr -msse2 -mfpmath=sse' -DCMAKE_INSTALL_PREFIX=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DLIB_DESTINATION=lib -DLIB_SUFFIX= -S . -B i586-alt-linux -DWITH_JBMC:BOOL=OFF -DBUILD_SHARED_LIBS:BOOL=OFF -Dsat_impl=system-minisat2
+ cmake --build i586-alt-linux --verbose --parallel 20
```

--- a/src/ansi-c/library_check.sh
+++ b/src/ansi-c/library_check.sh
@@ -15,7 +15,7 @@ for f in "$@"; do
     $CC -std=gnu11 -E -include library/cprover.h -D__CPROVER_bool=_Bool -D__CPROVER_thread_local=__thread -DLIBRARY_CHECK -o __libcheck.i __libcheck.c
     $CC -S -Wall -Werror -pedantic -Wextra -std=gnu11 __libcheck.i \
       -o __libcheck.s -Wno-unused-label -Wno-unknown-pragmas \
-      -Wno-gnu-line-marker -Wno-unknown-warning-option
+      -Wno-gnu-line-marker -Wno-unknown-warning-option -Wno-psabi
     ec="${?}"
     rm __libcheck.s __libcheck.i __libcheck.c
     [ "${ec}" -eq 0 ] || exit "${ec}"

